core/go/internal/publictxmgr/transaction_manager.go

```diff
@@ -775,8 +776,10 @@ func (ptm *pubTxManager) MatchUpdateConfirmedTransactions(ctx context.Context, d
 				// matched results in the order of the inputs
 				results = append(results, &components.PublicTxMatch{
 					PaladinTXReference: components.PaladinTXReference{
-						TransactionID:   match.Transaction,
-						TransactionType: match.TransactionType,
+						TransactionID:              match.Transaction,
+						TransactionType:            match.TransactionType,
+						TransactionSender:          match.Sender,
+						TransactionContractAddress: match.ContractAddress,
```

Comment:
This is where we propagate the original private smart contract being invoked from the binding, which is stashed at submission time.